### PR TITLE
BaSP-MR-95: Subscriptions routes and forms

### PR DESCRIPTION
**Pull request by JuanManuel-Blois**
Added subscriptions routes and updated the form component

### CODE REVIEW

File: src/Components/Subscriptions/FormSubscriptions/index.js [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        });
+        setSubscriptions((currentSub) => {
+          const updateSub = [...currentSub];
+          updateSub[dataIndex] = updateSub.data;
+          return updateSub;
+        });
+        setResponseModal({
+          title: 'Succes!',
+          description: updatedSubscription.message,
+          isConfirm: false
```
Comment (Tilajc) [inline src/Components/Subscriptions/FormSubscriptions/index.js:165]: You don't need it any more

Comment (Tilajc) [inline src/Components/Subscriptions/FormSubscriptions/index.js]: you don't need it any more
```suggestion
         
```

File: src/Components/Subscriptions/FormSubscriptions/index.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        setSubscriptionCreated(true);
+      } else {
+        setSubscriptionCreated(false);
+        throw new Error(updatedSubscription.message);
+      }
+    } catch (error) {
+      setResponseModal({
+        title: 'ERROR!',
+        description: error.message,
+        isConfirm: false
```
Comment (Tilajc) [inline src/Components/Subscriptions/FormSubscriptions/index.js]: you don't need it any more
```suggestion
        
```

File: src/Components/Subscriptions/FormSubscriptions/index.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    setIsOpen(!isOpen);
+  };
+
+  return (
+    <form className={style.formSubscription}>
+      <Modal
+        title={responseModal.title}
+        desc={responseModal.description}
+        isOpen={isOpen}
+        confirmModal={responseModal.isConfirm}
```
Comment (Tilajc) [inline src/Components/Subscriptions/FormSubscriptions/index.js]: you don't need it any more
```suggestion
        
```

**Review by Tilajc — CHANGES_REQUESTED** [review]
well done! I left a little comments

**Comment by LucianoStradiot** [comment]
[Form Components - Subscription](https://trello.com/c/m5TVTXhH/95-form-components-subscription)

**Review by Tilajc — APPROVED** [review]
Good Job!

**Comment by Katairala** [comment]
Tested locally on https://joaco-megarocket-app.vercel.app/
Test Results: Passed 🌟

It looks really nice! Great work Juan Manuel!

**Pull request merged**